platform-core/src/main/kotlin/org/dashevo/platform/Names.kt

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright (c) 2020-present, Dash Core Group
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+package org.dashevo.platform
+
+import org.bitcoinj.core.Base58
+import org.bitcoinj.core.ECKey
+import org.bitcoinj.core.Sha256Hash
+import org.bitcoinj.params.EvoNetParams
+import org.dashevo.dapiclient.DapiClient
+import org.dashevo.dapiclient.model.DocumentQuery
+import org.dashevo.dpp.document.Document
+import org.dashevo.dpp.identity.Identity
+import org.dashevo.dpp.util.Entropy
+import org.json.JSONObject
+import java.io.ByteArrayOutputStream
+import java.lang.Thread.sleep
+
+class Names (val platform: Platform) {
+
+    companion object {
+        const val DEFAULT_PARENT_DOMAIN = "dash"
+    }
+
+    fun register(name: String, identity: Identity, identityHDPrivateKey: ECKey): Document
+    {
+        val dpp = platform.dpp
+
+        val identityType = if (identity.type.value == 2) "application" else "user"
+
+        // @ts-ignore
+
+        val records = HashMap<String, Any?>(1)
+        records["dashIdentity"] = identity.id
+
+        val nameSlice = name.indexOf('.')
+        val normalizedParentDomainName =
+            if(nameSlice == -1) "dash" else name.slice(nameSlice + 1 .. name.length)
+
+        val label = if(nameSlice == -1) name else name.slice(0 .. nameSlice)
+
+        val normalizedLabel = label.toLowerCase();
+        val fullDomainName = "$normalizedLabel.$normalizedParentDomainName";
+
+        val nameHash = Sha256Hash.twiceOf(fullDomainName.toByteArray())
+        val nameHashHex = nameHash.toString()
+
+        val preorderSaltBase58 = Entropy.generate();
```

Comment:
Can we adjust so `preOrder` has the same casing it the two `val`s?

Reply:
`preorder` is one word, that is why it is all lowercase.